hash_practice/exercises.py

```diff
@@ -22,8 +51,76 @@ def valid_sudoku(table):
         Each element can either be a ".", or a digit 1-9
         The same digit cannot appear twice or more in the same 
         row, column or 3x3 subgrid
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n^2) but I'm not sure. 

```

Comment:
In this case we can actually say that it takes O(1)! This is because we are hard coded to a 9x9 grid, so it's always going to be the same.

If we allowed the grid to be any size it would be a bit more complicated depending on how you decided to handle it, so I can't give a good value for you there.